core/src/main/c/share/jit/x86.h

```diff
@@ -416,6 +432,36 @@ namespace questdb::x86 {
         }
     }
 
+    Gpq convert_to_int64(Compiler &c, const jit_value_t &val) {
+        switch (val.dtype()) {
+            case data_type_t::i8:
+            case data_type_t::i16:
+            case data_type_t::i32:
+                return int32_to_int64(c, val.gp().r32(), false /* checkNull */);
+            default:
+                return val.gp().r64();
+        }
+    }
+
+    jit_value_t to128(Compiler &c, const jit_value_t &lhs, const jit_value_t &rhs) {
+        c.comment("to_long128");
+
+        Gpq hi = convert_to_int64(c, lhs);
+        Gpq lo = convert_to_int64(c, rhs);
+
+        Mem mem = c.newStack(16, 16);
```

Comment:
without memory access
```
to128(long, long):
        movq    xmm0, rsi
        movq    xmm1, rdi
        punpcklqdq      xmm0, xmm1
        ret
```

Reply:
```
avx2_to128(long, long):
        vmovq   xmm0, rdi
        vmovq   xmm1, rsi
        vpunpcklqdq     xmm0, xmm1, xmm0        # xmm0 = xmm1[0],xmm0[0]
        ret
```

Reply:
Ah cool, I had been looking for a more efficient way to assemble `to_long128` and hadn't come up with a way that doesn't use memory. Nice!

I'm not sure I can use your `avx2_to128` though without significant changes to the avx2 compiler? I think to128 is a special case of using mixed types, in that you have to have 64-bit integers before to call to128 and you have to have 128-bit integers afterward. Maybe I'm missing something though

Reply:
Took the suggestion for `x86_to128`, leaving this thread active for the question about `avx2_to128`